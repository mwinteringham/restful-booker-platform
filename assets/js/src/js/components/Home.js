import React from 'react';
import { useState, useEffect } from 'react';
import HotelRoomInfo from './HotelRoomInfo.js';
import HotelMap from './HotelMap.js';
import HotelLogo from './HotelLogo.js';
import HotelContact from './HotelContact.js';

import { API } from '../libs/Api.js';

const Home = () => {

    const [rooms, setRooms] = useState([]);
    const [branding, setBranding] = useState({});

    useEffect(() => {
        API.getBranding(setBranding);
        API.getRoom(setRooms);
    }, []);
 
    return (
        <div>
            <div className="container-fluid">
                <div className="row">
                    <div className='col-sm-12 text-center'>
                        <HotelLogo logoDetails={branding.logoUrl} />
                    </div>
                </div>
                <div className="row hotel-description">
                    <div className='col-sm-1'></div>
                    <div className='col-sm-10'>
                        <p>{branding.description}</p>
                    </div>
                    <div className='col-sm-1'></div>
                </div>
                <div className="row room-header">
                    <div className='col-sm-1'></div>
                    <div className='col-sm-10'>
                        <h2>Rooms</h2>
                    </div>
                    <div className='col-sm-1'></div>
                </div>
                {rooms.map((roomDetails) => {
                    return <div key={roomDetails.roomid}><HotelRoomInfo roomDetails={roomDetails} /></div>
                })}
                    <HotelContact contactDetails={branding.contact} />
                <div className="row">
                    <div className='col-sm-12'>
                        <HotelMap name={branding.name} mapDetails={branding.map} />
                    </div>
                </div>
            </div>
        </div>
    );
}

export default Home;