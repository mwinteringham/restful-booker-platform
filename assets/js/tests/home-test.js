import React from 'react';
import Home from '../src/js/components/Home.js';
import nock from 'nock';

import '@testing-library/jest-dom'
import {
    render,
    waitFor
  } from '@testing-library/react'

const homeState = {
    rooms : [{
        roomid: 1,
        roomName: "101",
        type: 'Standard Room',
        accessible: false,
        image : 'https://www.mwtestconsultancy.co.uk/img/room1.jpg',
        description: 'Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.',
        features: [
            'Internet/Wi-fi',
            'Jacuzzi Bathroom',
            'Air conditioning',
            'High Definition TV',
            'Mini-bar'
        ]
    },{
        roomid: 2,
        roomName: "102",
        type: 'Single',
        accessible: true,
        image: 'https://www.mwtestconsultancy.co.uk/img/room2.jpg',
        description: 'Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.',
        features: [
            'Internet/Wi-fi',
            'Air conditioning',
            'Mini-bar'
        ]
    }],
    name : 'Shady meadows B&B',
    map : {
        latitude : 52.6351204,
        longitude : 1.2733774
    },
    logoUrl : 'https://www.mwtestconsultancy.co.uk/img/rbp-logo.png',
    description : "Welcome to Shady Meadows, a delightful Bed & Breakfast nestled in the hills on Newfordburyshire. A place so beautiful you will never want to leave. All our rooms have comfortable beds and we provide breakfast from the locally sourced supermarket. It's a delightful place.",
    contact : {
        name : 'Shady Meadows B&B',
        address : 'The Old Farmhouse, Shady Street, Newfordburyshire, NE1 410S',
        phone : '0123456789',
        email : 'fake@fakeemail.com'
    }
}

nock('http://localhost')
    .get('/room/')
    .reply(200, homeState);

nock('http://localhost')
    .get('/branding/')
    .reply(200, {
        name: 'A new name',
        map: {
            latitude: 88.123,
            longitude: 11.123
        },
        logoUrl: 'https://www.mwtestconsultancy.co.uk/url/update.png',
        description: 'Updated description',
        contact: {
            name: 'Another B&B',
            address: 'Somewhere else',
            phone: '99999999999',
            email: 'another@fakeemail.com'
        }
    });

test('Home page renders', async () => {
    const {getByText, asFragment} = render(
        <Home />
    )

    await waitFor(() => expect(getByText(/Updated description/)).toBeInTheDocument());
    expect(asFragment()).toMatchSnapshot();
});
