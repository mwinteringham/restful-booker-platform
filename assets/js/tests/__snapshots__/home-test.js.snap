// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-12 text-center"
        >
          <img
            alt="Hotel logoUrl"
            class="hotel-logoUrl"
            src="https://www.mwtestconsultancy.co.uk/url/update.png"
          />
        </div>
      </div>
      <div
        class="row hotel-description"
      >
        <div
          class="col-sm-1"
        />
        <div
          class="col-sm-10"
        >
          <p>
            Updated description
          </p>
        </div>
        <div
          class="col-sm-1"
        />
      </div>
      <div
        class="row room-header"
      >
        <div
          class="col-sm-1"
        />
        <div
          class="col-sm-10"
        >
          <h2>
            Rooms
          </h2>
        </div>
        <div
          class="col-sm-1"
        />
      </div>
      <div>
        <div>
          <div
            class="row hotel-room-info"
          >
            <div
              class="col-sm-1"
            />
            <div
              class="col-sm-3"
            >
              <img
                alt="Preview image of room101"
                class="img-responsive hotel-img"
                src="https://www.mwtestconsultancy.co.uk/img/room1.jpg"
              />
            </div>
            <div
              class="col-sm-7"
            >
              <h3>
                Standard Room
              </h3>
              <p>
                Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.
              </p>
              <ul>
                <li>
                  Internet/Wi-fi
                </li>
                <li>
                  Jacuzzi Bathroom
                </li>
                <li>
                  Air conditioning
                </li>
                <li>
                  High Definition TV
                </li>
                <li>
                  Mini-bar
                </li>
              </ul>
              <button
                class="btn btn-outline-primary float-right openBooking"
                type="button"
              >
                Book this room
              </button>
            </div>
            <div
              class="col-sm-1"
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="row hotel-room-info"
          >
            <div
              class="col-sm-1"
            />
            <div
              class="col-sm-3"
            >
              <img
                alt="Preview image of room102"
                class="img-responsive hotel-img"
                src="https://www.mwtestconsultancy.co.uk/img/room2.jpg"
              />
            </div>
            <div
              class="col-sm-7"
            >
              <span
                class="fa fa-wheelchair wheelchair"
              />
              <h3>
                Single
              </h3>
              <p>
                Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.
              </p>
              <ul>
                <li>
                  Internet/Wi-fi
                </li>
                <li>
                  Air conditioning
                </li>
                <li>
                  Mini-bar
                </li>
              </ul>
              <button
                class="btn btn-outline-primary float-right openBooking"
                type="button"
              >
                Book this room
              </button>
            </div>
            <div
              class="col-sm-1"
            />
          </div>
        </div>
      </div>
      <div
        class="row contact"
      >
        <div
          class="col-sm-1"
        />
        <div
          class="col-sm-5"
        >
          <form>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="basic-addon1"
                >
                  <span
                    class="fa fa-id-card"
                  />
                </span>
              </div>
              <input
                aria-describedby="basic-addon1"
                aria-label="Name"
                class="form-control"
                data-testid="ContactName"
                id="name"
                placeholder="Name"
                type="text"
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="basic-addon1"
                >
                  <span
                    class="fa fa-envelope"
                  />
                </span>
              </div>
              <input
                aria-describedby="basic-addon1"
                aria-label="Email"
                class="form-control"
                data-testid="ContactEmail"
                id="email"
                placeholder="Email"
                type="text"
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="basic-addon1"
                >
                  <span
                    class="fa fa-phone"
                  />
                </span>
              </div>
              <input
                aria-describedby="basic-addon1"
                aria-label="Phone"
                class="form-control"
                data-testid="ContactPhone"
                id="phone"
                placeholder="Phone"
                type="text"
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="basic-addon1"
                >
                  <span
                    class="fa fa-envelope"
                  />
                </span>
              </div>
              <input
                aria-describedby="basic-addon1"
                aria-label="Subject"
                class="form-control"
                data-testid="ContactSubject"
                id="subject"
                placeholder="Subject"
                type="text"
              />
            </div>
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  Message
                </span>
              </div>
              <textarea
                aria-label="Description"
                class="form-control"
                data-testid="ContactDescription"
                id="description"
                rows="5"
              />
            </div>
            <br />
            <button
              class="btn btn-outline-primary float-right"
              id="submitContact"
              type="button"
            >
              Submit
            </button>
          </form>
        </div>
        <div
          class="col-sm-5"
        >
          <p>
            <span
              class="fa fa-home"
            />
             Another B&B
          </p>
          <p>
            <span />
             Somewhere else
          </p>
          <p>
            <span
              class="fa fa-phone"
            />
             99999999999
          </p>
          <p>
            <span
              class="fa fa-envelope"
            />
             another@fakeemail.com
          </p>
        </div>
        <div
          class="col-sm-1"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-12"
        >
          <div
            class="map"
          >
            <div
              style="width: 1024px; height: 600px; position: relative; display: inline-block; overflow: hidden; background: rgb(221, 221, 221);"
            >
              <div
                class="pigeon-tiles-box"
                style="width: 1024px; height: 600px; position: absolute; overflow: hidden; will-change: transform; transform: scale(1, 1); transform-origin: top left;"
              >
                <div
                  class="pigeon-tiles"
                  style="position: absolute; width: 1280px; height: 768px; will-change: transform; transform: translate(-194.74204444512725px, -136.53495117649436px);"
                />
              </div>
              <div
                class="pigeon-overlays"
                style="position: absolute; width: 1024px; height: 600px;"
              >
                <div
                  class="pigeon-click-block"
                  style="position: absolute; transform: translate(497px, 269px); cursor: default;"
                >
                  <img
                    alt=""
                    height="34"
                    src="test-file-stub"
                    width="29"
                  />
                </div>
              </div>
              <div
                class="pigeon-attribution"
                style="position: absolute; bottom: 0px; right: 0px; font-size: 11px; padding: 2px 5px; background: rgba(255, 255, 255, 0.7); font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; color: rgb(51, 51, 51);"
              >
                <span>
                  <a
                    href="https://pigeon-maps.js.org/"
                    rel="noreferrer noopener"
                    style="color: rgb(0, 120, 168); text-decoration: none;"
                    target="_blank"
                  >
                    Pigeon
                  </a>
                   | 
                </span>
                <span>
                   © 
                  <a
                    href="https://www.openstreetmap.org/copyright"
                    rel="noreferrer noopener"
                    style="color: rgb(0, 120, 168); text-decoration: none;"
                    target="_blank"
                  >
                    OpenStreetMap
                  </a>
                   contributors
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
